src/test_suite/multiprocessing_utils.py

```diff
@@ -68,11 +71,42 @@ def read_context_serialized(test_file: Path) -> str | None:
     """
 
     # Serialize instruction context to string (pickleable)
-    ctx = read_context(test_file)
+    ctx = read_context(harness_ctx, test_file)
     return ctx.SerializeToString(deterministic=True) if ctx else None
 
 
-def read_context(test_file: Path) -> message.Message | None:
+def extract_metadata(fixture_file: Path) -> str | None:
+    """
+    Extracts metadata from a fixture file.
+
+    Args:
+        - fixture_file (Path): Path to the fixture message.
+
+    Returns:
+        - str | None: Metadata from the fixture file.
+    """
+
+    with open(fixture_file, "rb") as f:
+        proto_bytes = f.read()
+        try:
+            metadata = metadata_pb2.FixtureMetadata()
+            pos = 0
+            while pos < len(proto_bytes):
+                tag, pos = _DecodeVarint(proto_bytes, pos)
+                if (tag >> 3) == 1:
+                    length, pos = _DecodeVarint(proto_bytes, pos)
+                    metadata.ParseFromString(proto_bytes[pos : pos + length])
+                    return metadata
+                pos += _DecodeVarint(proto_bytes, pos)[0]
+            raise message.DecodeError("No 'metadata' found")
+        except message.DecodeError as e:
+            print(f"Failed to parse 'metadata': {e}")
+            return None
+
+    return getattr(metadata, "fn_entrypoint", None)
```

Comment:
why return this instead of just metadata?

Reply:
unreachable line